manifest/api/models/huggingface.py

```diff
@@ -646,3 +673,129 @@ def score_sequence(
+            **kwargs,
+        )
+        return encoded_prompt
+
+    @torch.no_grad()
+    def score_sequence_eleuther_lm_eval(
+        self, 
+        prompt_with_label: List[Tuple[str, str]], 
+        **kwargs: Any
+    ) -> List[Tuple[str, str, float]]:
+        """
+        Score a list of (prompt + label) the way the EleutherAI eval harness does.
+
+        Args:
+            prompt_with_label: List[Tuple[str,str]]
+                The prompt to score the labels against,
+                plus its corresponding label.
+                Unlike the other `score_sequence()` method,
+                the label is not included in the prompt.
+            **kwargs:
+                Additional keyword arguments passed along to the :obj:`__call__` method.
+        """
+        prompts: List[str] = [x[0] for x in prompt_with_label]
+        labels: List[str] = [x[1] for x in prompt_with_label]
+
+        if self.model_type == "text2text-generation":
+            # For seq2seq models, we add the label as the
+            # output expected from the decoder, and separate
+            # the label from the prompt (input to the encoder)
+            encoded_prompt = self.pipeline.tokenizer(
+                prompts,
+                max_length=self.pipeline.max_length,
+                truncation=True,
+                padding=True,
+                add_special_tokens=False,
+                return_tensors="pt",
+            )
+            encoded_prompt["labels"] = self.pipeline.tokenizer(
+                labels,
+                max_length=self.pipeline.max_length,
+                return_tensors="pt",
+                truncation=True,
+                padding=True,
+                add_special_tokens=False,
+            )["input_ids"]
+            encoded_prompt = encoded_prompt.to(self.pipeline.device)
+            # Generate model logits for `labels`
+            outputs = self.pipeline.model(**encoded_prompt)  # type: ignore
+            log_softmaxes = torch.log_softmax(outputs.logits, dim=-1)
+            label_mask = (
+                encoded_prompt["labels"] != self.pipeline.tokenizer.pad_token_id
+            )
+            label_token_probs = torch.gather(
+                log_softmaxes, -1, encoded_prompt["labels"].unsqueeze(-1)
+            ).squeeze(-1)
+            masked_probs = label_mask * label_token_probs
+            label_probs = masked_probs.sum(dim=-1)
+        elif self.model_type == "text-generation":
+            # For gpt2-like models, we append the label to the
+            # end of the prompt, measure the entire sequence's
+            # logprob, and count that as the label's logprob
+            # (since the label is the only thing that changes
+            # between sequences, so the prompt's logprob
+            # will be constant across all labels)
+
+            # Pad right instead of left
+            encoded_prompt = self.pipeline.tokenizer(
+                [f"{x} {y}" for (x, y) in zip(prompts, labels)],
```

Comment:
Does this assume no space on the labels? I've played with this method before and found it super sensitive to spacing of the labels? Not sure if it's a good idea to strip the labels?